python/pyarrow/includes/libarrow.pxd

```diff
@@ -275,6 +281,15 @@ cdef extern from "arrow/api.h" namespace "arrow" nogil:
     cdef cppclass CStringArray" arrow::StringArray"(CBinaryArray):
         c_string GetString(int i)
 
+    cdef cppclass CStructArray" arrow::StructArray"(CArray):
+        CStructArray(shared_ptr[CDataType] type, int64_t length,
+            vector[shared_ptr[CArray]] children,
+            shared_ptr[CBuffer] null_bitmap = nullptr, int64_t null_count = 0,
```

Comment:
You'll probably need to put `from libcpp cimport nullptr` at the top of `arrow/python/pyarrow/includes/common.pxd` for this to work.